src/govuk/common.mjs

```diff
@@ -192,8 +204,8 @@ export function normaliseDataset (dataset) {
  * Returns the value of the given attribute closest to the given element (including itself)
  *
  * @param {HTMLElement} $element - The element to start walking the DOM tree up
- * @param {String} attributeName - The name of the attribute
- * @returns {String|undefined}
+ * @param {string} attributeName - The name of the attribute
+ * @returns {string | null} Attribute value
```

Comment:
Although getAttribute can return `null`, I don't think it's a logical possibility here because we've used an attribute selector which means the attribute _must_ exist.

However, if there is no element with that attribute then there is no explicit return from the function which I think means it would return `undefined` rather than `null`?

Reply:
I do appreciate the through reviews, thank you for this. I've got into a habit of using the TypeScript language service to check the return types everywhere which flagged this up

Like you say, it's correct that **getAttribute()** returns `string | null`, but it's not clever enough to know we're selecting the same attribute above 🤦‍♂️ 

I think JSDoc 3.2+ supports `{string | void}` (nothing returned) but I'll put `undefined` back as it's still valid

```suggestion
 * @returns {string | undefined} Attribute value
```